tracer/src/Datadog.Trace/Logging/Internal/DatadogLogging.cs

```diff
@@ -207,5 +209,30 @@ internal static string GetLogDirectory()
 
             return logDirectory;
         }
+
+        internal static void CleanLogFiles(string logsDirectory)
+        {
+            DirectoryInfo logsDirInfo = new DirectoryInfo(logsDirectory);
+            FileInfo[] logFiles = logsDirInfo.GetFiles().OrderBy(p => p.LastWriteTime).ToArray();
+
+            for (int i = 0; i < logFiles.Length; i++)
+            {
+                if ((DateTime.Now - logFiles[i].LastWriteTime).TotalDays >= 31)
+                {
+                    try
+                    {
+                        logFiles[i].Delete();
+                    }
+                    catch
+                    {
+                        // Do nothing when an exception is thrown for attempting to delete the filesystem
```

Comment:
Any IO, like `DirectoryInfo.GetFiles()` can throw an exception, so we should probably also wrap the entire method in a try/catch.

Reply:
But also each individual file in case one fails we can still delete the others.